internal/component/database_observability/postgres/collector/query_samples.go

```diff
@@ -137,6 +142,26 @@ type SampleState struct {
 	LastSeenAt  time.Time
 	LastCpuTime string // last cpu_time observed under CPU condition
 	tracker     WaitEventTracker
+	// EndOverride is used to compute durations and timestamps when a query
+	// transitioned to idle or was only observed as idle.
+	EndOverride sql.NullTime
```

Comment:
This name is a bit confusing to me - it's not an override but when we determined the sample was idle?